flang/lib/Lower/OpenMP.cpp

```diff
@@ -3377,10 +3414,17 @@ static void genOMP(Fortran::lower::AbstractConverter &converter,
                               ")");
   }
 
-  // 2.9.3.1 SIMD construct
   if (llvm::omp::allSimdSet.test(ompDirective)) {
-    createSimdLoop(converter, eval, ompDirective, loopOpClauseList,
-                   currentLocation);
+    if (isWorkshareSimdConstruct(ompDirective)) {
```

Comment:
We can use one of the existing directive sets here instead:
```suggestion
    if (llvm::omp::allDoSet.test(ompDirective)) {
```
We could also remove the need for a nested 'if' as well:
```c++
if (llvm::omp::allDoSimdSet.test(ompDirective))
  createSimdWsLoop(...);
else if (llvm::omp::allSimdSet.test(ompDirective))
  createSimdLoop(...);
else
  createWsLoop(...);
```

Reply:
Thanks for the remark. I applied it.

Reply:
Thanks!